### PR TITLE
Update change log for 6.1.0-preview.1

### PR DESCRIPTION
**Pull request by daxian-dbw**
## PR Summary

Update change log for 6.1.0-preview.1

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,160 @@
...
+### Engine Updates and Fixes
+
+- Fix `ValidateSet` with generator in a module (#5702)
+- Update `SAL` annotation and fix warnings (#5617)
+- Add `ForEach` and `Where` methods to `PSCustomobject` (#5756) (Thanks @iSazonov!)
+- Add `Count` and `Length` properties to `PSCustomobject` (#5745) (Thanks @iSazonov!)
+- Make minor fixes in compiler to properly handle void type expression (#5764)
+- Logging: Fix the escaped characters when generating `.resx` file from PowerShell `ETW` manifest. (#5892)
+- Remove `PSv2` only code from `Types_Ps1Xml.cs` and `HostUtilities.cs` (#5907) (Thanks @iSazonov!)
+- Do not pass `ArgumentAst` to `CreateArgument` for `pwsh -encodedarguments` (#5836)
```
Comment (SteveL-MSFT) [inline CHANGELOG.md]: This description isn't useful to end users.  Perhaps: Enable passing arrays to `pwsh -EncodedArguments` on debug builds.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,160 @@
...
+- Logging: Handle path that contains spaces in `RegisterManifest.ps1` (#5859) (Thanks @tandasat!)
+- Add `-settingsfile` to `pwsh` to support loading a custom powershell config file. (#5920)
+- Return better error for `pwsh -WindowStyle` on unsupported platforms. (#5975) (Thanks @thezim!)
+- Enable conversions from `PSMethod` to `Delegate` (#5287) (Thanks @powercode!)
+- Minor code clean-up changes in tab completion code (#5737) (Thanks @kwkam!)
+- Add lambda support to `-replace` operator (#6029) (Thanks @IISResetMe!)
+- Fix error in windows provider when the environment as an existing set of variables name that only differs by case (#6320)
```
Comment (SteveL-MSFT) [inline CHANGELOG.md]: Title is confusion.  Use: Fix retrieval of environment variables on Windows in cases where variable names differ only by case.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,160 @@
...
+- Fix the filtering of analytic events on Unix platforms. (#6086)
+- Update copyright and license headers (#6134)
+- Set pipeline thread stack size to 10MB (#6224) (Thanks @iSazonov!)
+
+### General Cmdlet Updates and Fixes
+
+- Fix the `NullRefException` in `Enter-PSHostProcess` (#5995)
+- Merge and Sort `BasicHtmlWebResponseObject` and `ContentHelper` in Web Cmdlets (#5720) (Thanks @markekraus!)
+- Encoding for `New-ModuleManifest` on all platforms should be `UTF-8 NoBOM` (#5923)
+- Make `Set-Location` use path with wildcard characters if it exists (#5839)
```
Comment (SteveL-MSFT) [inline CHANGELOG.md]: Make `Set-Location` use path with wildcard characters as literal if it exists

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,160 @@
...
+- Make `UTF-8` the default encoding for `application/json` (#6109) (Thanks @markekraus!)
+- Enable `$env:PAGER` to work correctly if arguments are used (#6144)
+- Remove the unnecessary `AddTypeCommandBase` class (#5407) (Thanks @iSazonov!)
+
+### Test
+
+- Convert Web Cmdlets test to `one-true-brace-style` formatting (#5716) (Thanks @markekraus!)
+- Add a test for `IValidateSetValuesGenerator` in a module (#5830) (Thanks @iSazonov!)
+- Fix function to test for docker OS due to change to use `linuxkit` for macOS (#5843)
+- Replace `HttpListener` link header tests with `WebListener` (#5806) (Thanks @markekraus!)
```
Comment (SteveL-MSFT) [inline CHANGELOG.md]: I think we can squash these three with just: Replace `HttpListener` tests with `WebListener`

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,160 @@
...
+- Replace `HttpListener` echo tests with `WebListener` (#5840) (Thanks @markekraus!)
+- Replace `HttpListener` redirect tests with `WebListener` (#5872) (Thanks @markekraus!)
+- Stop `HttpListener` from running in Web Cmdlets tests (#5921) (Thanks @markekraus!)
+- Fix `PSVersion` in `PSSessionConfiguration` tests (#5554) (Thanks @iSazonov!)
+- Update test framework and tests to support Pester version 4 (#6064)
```
Comment (SteveL-MSFT) [inline CHANGELOG.md]: spelling out `version` is odd, how about just `Pester v4`?

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,160 @@
...
+- Update tests in `Microsoft.PowerShell.Management` folder to use Pester 4 Syntax. (#6294)
```
Comment (SteveL-MSFT) [inline CHANGELOG.md]: I think we should squash the Pester v4 syntax ones to just "Update tests to use Pester v4 syntax" listing out all the PR #s

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,160 @@
...
+- Change Web Cmdlet tests to use `127.0.0.1` instead of `Localhost` (#6069) (Thanks @markekraus!)
+- Fix `Start-PSPester` to include or exclude `RequireSudoOnUnix` tag smartly on Unix (#6241)
+- Fix the terse output on Windows for test runs without admin privilege (#6252)
+- Add `RequireSudoOnUnix` tag for `Get-Help` tests. (#6223)
+- Add tests for `*-Item` Cmdlets in function provider (#6172)
+- Support running tests in root privilege on Linux. (#6145)
+
+### Build and Packaging Improvements
+
+- Add option to add shell explorer context menu in Windows installer (#5774) (Thanks @bergmeister!)
```
Comment (SteveL-MSFT) [inline CHANGELOG.md]: I think it should be "explorer shell" rather than "shell explorer"

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,160 @@
...
+- Make the shell explorer context menu registry entries platform specific to allow side by side of x86 and `x64`. (#5824) (Thanks @bergmeister!)
```
Comment (SteveL-MSFT) [inline CHANGELOG.md]: Should be consistent with enclosing x86 in backticks.  Also should be `explorer shell`

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,160 @@
...
+- Update docs with test guidelines with the `RequireSudoOnUnix` tag. (#6274)
+- Add `Alpine` Linux support (#6367) (Thanks @kasper3!)
+- Update to Governance doc to reflect current working model (#6323)
+- Add guidance on adding copyright and license header to new source files (#6140)
+- Fix the command to build type catalog in `internals.md` (#6084) (Thanks @ppadmavilasom!)
+- Fix `Pull Request Process` dead link (#6066) (Thanks @IISResetMe!)
+- Update processes to allow for coordinated vulnerability disclosure (#6042)
+- Rework Windows Start menu folder name (#5891) (Thanks @Stanzilla!)
+- Update `Raspbian` installation instructions to create `symlink` for `pwsh` (#6122)
+- fix various places that still refer to old versions of `pwsh` (#6179) (Thanks @bergmeister!)
```
Comment (TravisEz13) [inline CHANGELOG.md]: Capitalize first word for consistency

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,160 @@
...
+- Fix build, packaging and installation scripts for `SLES` (#5918) (Thanks @tomconte!)
+- Update recommended `WiX` toolset link to be generic to `WiX 3.x` but mention that latest version of 3.11 has to be taken (#5926) (Thanks @bergmeister!)
+- Add service point manager call in `Install-PowerShell.ps1` to force `TLS1.2`. (#6310) (Thanks @DarqueWarrior!)
+- Add `-Restore` when build `win-arm` and `win-arm64` (#6353)
+- Make sure package verification failure fails the `AppVeyor` build (#6337)
+- Specify the runtime when running `dotnet restore` in `Start-PSBuild` (#6345)
+- Rename `log` and `logerror` to `Write-Log [$message] [-error]` (#6333)
+- Make Linux packages use correct version scheme for preview releases (#6318)
+- Add support for Debian in `installpsh-debian.sh` (#6314) (Thanks @Pawamoy!)
+- MSI: make preview builds to install Side by side with release builds (#6301)
```
Comment (TravisEz13) [inline CHANGELOG.md]: `Make` should be capitalized

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,160 @@
 # Changelog
 
+## v6.1.0-preview.1 - 2018-03-23
+
+### Breaking Changes
+
+- Throw terminating error in `New-TemporaryFile` and make it not rely on the presence of the `TEMP` environment variable (#6182) (Thanks @bergmeister!)
+- Remove the unnecessary `AddTypeCommandBase` class from `Add-Type` (#5407) (Thanks @iSazonov!)
+- Remove unsupported members from the enum `Language` in `Add-Type` (#5829) (Thanks @iSazonov!)
+- Fix range operator to work better with character ranges (#5732) (Thanks @iSazonov!)
```
Comment (TravisEz13) [inline CHANGELOG.md]: The template for the announcements repo said that we would have an announcement for breaking changes.

**Comment by daxian-dbw** [comment]
Thanks @SteveL-MSFT and @TravisEz13 for your comments. They are all addressed except for the `announcement repo` one.

File: CHANGELOG.md (hunk 11 above)
Comment (SteveL-MSFT) [reply CHANGELOG.md]: We should keep `Breaking Changes` here, but should also separately post to `Announcements` repo

**Review by SteveL-MSFT — APPROVED** [review]

**Review by joeyaiello — APPROVED** [review]
Wow, this is a lot of changes! Especially the all the build/packaging stuff, you guys have been really busy. :) 

Changelog looks great!

**Pull request merged**